src/irr/asset/CImageWriterJPG.cpp

```diff
@@ -128,40 +192,25 @@ static bool writeJPEGFile(io::IWriteFile* file, const asset::CImageData* image,
 	jpeg_set_quality(&cinfo, quality, TRUE);
 	jpeg_start_compress(&cinfo, TRUE);
 
-	uint8_t * dest = new uint8_t[dim.X*3];
+	const auto JPG_BYTE_PITCH = texelOrBlockByteSize;
+	uint8_t* dest = new uint8_t[JPG_BYTE_PITCH];
 
 	if (dest)
 	{
-		const uint32_t pitch = image->getPitchIncludingAlignment();
+		const uint32_t pitch = JPG_BYTE_PITCH;
 		JSAMPROW row_pointer[1];      /* pointer to JSAMPLE row[s] */
 		row_pointer[0] = dest;
 		
-		uint8_t* src = (uint8_t*)image->getData();
+		uint8_t* src = (uint8_t*)convertedImage->getBuffer()->getPointer();
 		
 		/* Switch up, write from bottom -> top because the texture is flipped from OpenGL side */
 		uint32_t eof = cinfo.image_height * cinfo.image_width * cinfo.input_components;
 		
 		while (cinfo.next_scanline < cinfo.image_height)
 		{
-			/* Since the first argument to convertColor() requires a const void *[4], wrap our buffer pointer (src) and pass that to convertColor(). */
-			const void *src_container[4] = {src, nullptr, nullptr, nullptr};
-			
-			/* Pass-through the pixels for EF_R8_SRGB/EF_R8G8B8_SRGB, and perform color conversion for the rest.*/
-			switch (format) {
-				case asset::EF_R8G8B8_UNORM:
-					video::convertColor<EF_R8G8B8_UNORM, EF_R8G8B8_SRGB>(src_container, dest, dim.X, dim);
-					break;
-				case asset::EF_B5G6R5_UNORM_PACK16:
-					video::convertColor<EF_B5G6R5_UNORM_PACK16, EF_R8G8B8_SRGB>(src_container, dest, dim.X, dim);
-					break;
-				case asset::EF_A1R5G5B5_UNORM_PACK16:
-					video::convertColor<EF_A1R5G5B5_UNORM_PACK16, EF_R8G8B8_SRGB>(src_container, dest, dim.X, dim);
-					break;
-				case asset::EF_R8_UNORM:
-					video::convertColor<EF_R8_UNORM, EF_R8_SRGB>(src_container, dest, dim.X, dim);
-					break;
-				case asset::EF_R8_SRGB:
-					_IRR_FALLTHROUGH;
+			switch (convertedFormat)
+			{
+				case asset::EF_R8_SRGB: _IRR_FALLTHROUGH;
 				case asset::EF_R8G8B8_SRGB:
 					memcpy(dest, src, pitch);
 					break;
```

Comment:
explain what's going on here

Reply:
those are variables and pointers which value depends on final format, but I left is simmilar as it was before

actually it just copy to a row needed for jpg library (and png does the same) for writing process, but it will warn if by some miracle format at that point isn's EF_R8_SRGB or EF_R8G8B8_SRGB

Reply:
it's good

Reply:
ok